File: 백준/Bronze/2530. 인공지능 시계/인공지능 시계.py
# 입력: 시, 분, 초 및 조리 시간(초)
# 출력: 종료되는 시각 시, 분, 초

hour, minute, second = map(int, input().split())
cooking_time = int(input())

# 1. 초 단위로 변환 
total_seconds = hour * 3600 + minute * 60 + second + cooking_time

# 2. 24시간 형식으로 변환
total_seconds %= 86400  # 하루는 86400초
end_hour = total_seconds // 3600
total_seconds %= 3600
end_minute = total_seconds // 60
end_second = total_seconds % 60

# 3. 결과 출력
print(end_hour, end_minute, end_second)

File: 백준/Bronze/2530. 인공지능 시계/인공지능 시계.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8').trim().split(/\s+/).map(Number);

let [hour, minute, second] = input.slice(0, 3);
let need_second = input[3];

second += Math.floor(need_second % 60);
if(second >= 60){
    minute += Math.floor(second / 60);
    second = second % 60;
}

minute += Math.floor(need_second / 60) % 60;
if(minute >= 60){
    hour += Math.floor(minute / 60);
    minute = minute % 60;
}

hour += Math.floor(need_second / 3600);
if(hour >= 24){
    hour = hour % 24;
}

console.log(`${hour} ${minute} ${second}`);